content/en/blog/2023/testing-otel-demo/index.md

```diff
@@ -0,0 +1,399 @@
+- _“The product was shipped”_, checking if the
+  [ShippingService](https://github.com/open-telemetry/opentelemetry-demo/tree/main/src/shippingservice)
+  was called and emitted spans correctly;
+- _“The cart was emptied”_, checking if the
+  [CartService](https://github.com/open-telemetry/opentelemetry-demo/tree/main/src/cartservice)
+  was called and emitted spans correctly.
+
+The final result was the following test YAML, which triggers the Checkout
+operation and validates these five groups of assertions:
+
+```yaml
+type: Test
+spec:
+  name: 'Frontend: Checking out shopping cart'
+  description: Simulate user checking out shopping cart
+  trigger:
+    type: http
+    httpRequest:
+      url: http://{{frontend address}}/api/checkout
+      method: POST
+      headers:
+      - key: Content-Type
+        value: application/json
+      body: |
+        {
+          "userId": "2491f868-88f1-4345-8836-d5d8511a9f83",
+          "email": "someone@example.com",
+          "address": {
+            "streetAddress": "1600 Amphitheatre Parkway",
+            "state": "CA",
+            "country": "United States",
+            "city": "Mountain View",
+            "zipCode": "94043"
+          },
+          "userCurrency": "USD",
+          "creditCard": {
+            "creditCardCvv": 672,
+            "creditCardExpirationMonth": 1,
+            "creditCardExpirationYear": 2030,
+            "creditCardNumber": "4432-8015-6152-0454"
+          }
+        }
+  specs:
+  - name: "The frontend has been called with success"
+    selector: span[name="Test trigger"]
+    assertions:
+    - attr:response.status = 200
+  - selector: span[name="oteldemo.CheckoutService/PlaceOrder" rpc.system="grpc"
+			rpc.method="PlaceOrder" rpc.service="oteldemo.CheckoutService"]
+    name: "The order was placed"
+    assertions:
+    - attr:app.user.id = "2491f868-88f1-4345-8836-d5d8511a9f83"
+    - attr:app.order.items.count = 1
+  - selector: span[name="oteldemo.PaymentService/Charge" rpc.system="grpc"
+			rpc.method="Charge" rpc.service="oteldemo.PaymentService"]
+    name: "The user was charged"
+    assertions:
+    - attr:rpc.grpc.status_code  =  0
+    - attr:selected_spans.count >= 1
+  - selector: span[name="oteldemo.ShippingService/ShipOrder" rpc.system="grpc"
+			rpc.method="ShipOrder" rpc.service="oteldemo.ShippingService"]
+    name: "The product was shipped"
+    assertions:
+    - attr:rpc.grpc.status_code = 0
+    - attr:selected_spans.count >= 1
+  - selector: span[name="oteldemo.CartService/EmptyCart" rpc.system="grpc"
+			rpc.method="EmptyCart" rpc.service="oteldemo.CartService"]
+    name: "The cart was emptied"
+    assertions:
+    - attr:rpc.grpc.status_code = 0
+    - attr:selected_spans.count >= 1
+```
+
+Finally, when running these tests we have the following report, showing each
+test file run in the transaction and also with the "Checkout” step described
+above:
+
+![results from the trace-based test](./trace-based-test-result.png)
```

Comment:
Since this is text only I recommend to make use of the text directly

Reply:
Agreed. Changing to text.